core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrCylinder.java

```diff
@@ -171,63 +178,43 @@ public Point3 getIntersectionInSource(Point3 rv, Ray ray, AffineMatrix4x4 m, int
     double t = Double.NaN;
     final double THRESHOLD = 0.01;
     if (HANDLE_CONES_SEPARATELY && (Math.abs(this.bottomRadius - this.topRadius) < THRESHOLD)) {
-      Vector3 originToOrigin = Vector3.createSubtraction(ray.accessOrigin(), cylinderPosition);
-      Vector3 rayDirection_X_cylinderDirection = Vector3.createCrossProduct(ray.accessDirection(), cylinderDirection);
+      Vector3 originToOrigin = ray.origin().minus(cylinderPosition);
+      Vector3 rayDirection_X_cylinderDirection = ray.direction().crossProduct(cylinderDirection);
 
-      double magnitude = rayDirection_X_cylinderDirection.calculateMagnitude();
+      double magnitude = rayDirection_X_cylinderDirection.magnitude();
 
       if (magnitude > EpsilonUtilities.REASONABLE_EPSILON) {
-        rayDirection_X_cylinderDirection.normalize();
-        double d = Math.abs(Vector3.calculateDotProduct(originToOrigin, rayDirection_X_cylinderDirection));
+        rayDirection_X_cylinderDirection = rayDirection_X_cylinderDirection.normalized();
+        double d = Math.abs(originToOrigin.dotProduct(rayDirection_X_cylinderDirection));
         if (d <= maxRadius) {
-          Vector3 originToOrigin_X_cylinderDirection = Vector3.createCrossProduct(originToOrigin, cylinderDirection);
-          double a = -Vector3.calculateDotProduct(originToOrigin_X_cylinderDirection, rayDirection_X_cylinderDirection) / magnitude;
+          Vector3 originToOrigin_X_cylinderDirection = originToOrigin.crossProduct(cylinderDirection);
+          double a = -originToOrigin_X_cylinderDirection.dotProduct(rayDirection_X_cylinderDirection) / magnitude;
 
-          Vector3 rayDirection_X_CylinderDirection___X_cylinderDirection = Vector3.createCrossProduct(rayDirection_X_cylinderDirection, cylinderDirection);
-          rayDirection_X_CylinderDirection___X_cylinderDirection.normalize();
-          double b = Math.abs(Math.sqrt((maxRadius * maxRadius) - (d * d)) / Vector3.calculateDotProduct(ray.accessDirection(), rayDirection_X_CylinderDirection___X_cylinderDirection));
+          Vector3 rayDirection_X_CylinderDirection___X_cylinderDirection = rayDirection_X_cylinderDirection.crossProduct(cylinderDirection);
+          rayDirection_X_CylinderDirection___X_cylinderDirection = rayDirection_X_CylinderDirection___X_cylinderDirection.normalized();
+          double b = Math.abs(Math.sqrt((maxRadius * maxRadius) - (d * d)) / ray.direction().dotProduct(rayDirection_X_CylinderDirection___X_cylinderDirection));
 
           t = a - b;
         }
       }
     } else {
       //todo handle cones
     }
-    if (Double.isNaN(t)) {
-      rv.setNaN();
-    } else {
-      ray.getPointAlong(rv, t);
-    }
+    Point3 p = Double.isNaN(t) ? Point3.NaN : ray.getPointAlong(t);
 
-    if (rv.isNaN()) {
+    if (p.isNaN()) {
       //todo: check to see if hit cap
```

Comment:
is this comment still accurate? it seem like we're capping just fine?

Reply:
The comment was in the earliest version of this class and the logic has not changed much in this part, so it means as much as it did then.
This comment is so clear a 4 year old could understand it. Go get me a 4 year old, I can't make heads or tails of it.

Reply:
i think we can safely remove it